runtimes/runtimes/standalone.ts

```diff
@@ -89,9 +90,37 @@ if (checkAWSConfigFile()) {
     process.env.AWS_SDK_LOAD_CONFIG = '1'
 }
 
-process.on('uncaughtExceptionMonitor', err => {
-    console.error('Uncaught Exception:', err.message)
-})
+// Set up crash monitoring to emit telemetry before process exits
+function setupCrashMonitoring(telemetryEmitter?: (metric: MetricEvent) => void) {
+    function getTopStackFrames(err: Error): string | undefined {
+        if (!err.stack) {
+            return undefined
+        }
+
+        const stackLines = err.stack.split('\n')
+        const topStackFrames = stackLines.slice(0, 10).join('\n')
+        return topStackFrames
+    }
+
+    process.on('uncaughtExceptionMonitor', (err, origin) => {
+        console.error('Uncaught Exception:', err.message, getTopStackFrames(err))
+
+        if (telemetryEmitter) {
+            try {
+                telemetryEmitter({
+                    name: 'runtime_processCrash',
+                    result: 'Failed',
+                    errorData: {
+                        reason: err.toString(),
```

Comment:
`reason` for a "code" or name (`(err as any).code ?? (err as any).name`).

`reasonDesc` is for details. And should use `getTelemetryReasonDesc` to avoid PII https://github.com/aws/language-servers/blob/a592b8b0dce29b485ab79def373e37ecec04b5ac/server/aws-lsp-codewhisperer/src/shared/utils.ts#L174